src/dune_sexp/versioned_file.ml

```diff
@@ -57,11 +57,13 @@ struct
 
     let parse first_line : Instance.t =
       let { First_line.lang = name_loc, name; version = ver_loc, ver } = first_line in
+      let ver_atom =
+        match Atom.parse ver with
+        | Some atom -> atom
+        | None -> User_error.raise ~loc:ver_loc [ Pp.text "Invalid version string" ]
```

Comment:
I think there is some code which can print the supported versions, it would be good to invoke that here (maybe `Syntax.check_supported`) because that allows people to see what they should set it to instead.

(You can trigger it if you set your dune-lang to a version like `(lang dune 4000.1)`)